pipeline/env/env.go

```diff
@@ -24,6 +24,10 @@ func (Pipe) Description() string {
 // Run the pipe
 func (Pipe) Run(ctx *context.Context) (err error) {
 	ctx.Token = os.Getenv("GITHUB_TOKEN")
+	if !ctx.Publish {
+		// The GITHUB_TOKEN is only needed during the publish phase.
```

Comment:
probably we will not publish snapshots too, so we may add that if here, or make `snapshot` also set publish to false...